docs/en/sql-reference/functions/time-series-functions.md

```diff
@@ -170,3 +170,61 @@ Result:
     ]]                                                                                                                   │
 └────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
 ```
+
+## seriesDecomposeAnomaliesDetection
+
+Detect anomalies in series data using [series decomposition](#seriesDecomposeSTL).
+
+**Syntax**
+
+``` sql
+seriesDecomposeAnomaliesDetection(series);
+seriesDecomposeAnomaliesDetection(series, threshold, seasonality, AD_method);
+```
+
+**Arguments**
+
+- `series` - An array of numeric values
+- `threshold` - A positive number to detect mild or stronger anomalies. The default is 1.5, K value.
+- `Seasonality` - Represents the period for the seasonal analysis of series data. Supported values are:
+                - `-1`: Autodetect period using [seriesPeriodDetectFFT](#seriesPeriodDetectFFT). The default value.
+                - `0`: Used to skip extracting seasonal component.
+                - A positive integer representing the period of series.
+- `AD_method` - The method to use for anomaly detection on residual component. Supported values are:
```

Comment:
`seriesDetectSTL` returns an array with four elements, each of which is a series.

If I understand correctly, we are interested in the residual component (the third array element).

Do we really need a new SQL function at all or can the detection logic be achieved by nesting `seriesDetectSTL`, seriesPeriodDetectFFT`, `seriesDetectTukey` in SQL? That seems way more modular and flexible from a user perspective. In that case, I'd suggest to add a small tutorial/guide to the time series function docs which demonstrates the process.

Reply:
yes, we use seriesDecomposeSTL and are mainly interested in the residual component from the result.

The primary objective of `seriesDecomposeAnomaliesDetect` method is to detect anomalies in time series data. To achieve this, it utilizes the available auxiliary methods  - seriesPeriodDetectFFT, seriesDecomposSTL and seriesOutliersTukey. Surely these methods can be utilized independently too. However, when the main purpose of the end user is to analyze their time series data for any anomalies, I believe they shouldn't need to worry about the underlying methods or algorithms.